### PR TITLE
initial pipeline setup

### PR DESCRIPTION
**Pull request by jsyska**

### CODE REVIEW

**Review by jankrokos — APPROVED** [review]

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+    - name: Deploy ACR
+      id: deploy_acr
+      uses: azure/arm-deploy@v1
+      with:
+        subscriptionId: ${{ secrets.AZURE_SUBSCRIPTION }}
+        resourceGroupName: ${{ vars.AZURE_RG }}
+        template: ./.github/workflows/acr-template.json
+        parameters: acrName=${{ vars.AZURE_ACR_NAME}}
```
Comment (CreativePhilip) [inline .github/workflows/deploy.yml:26]: Pls make formatting consistent

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    - name: Deploy App Service and Web App
+      uses: azure/arm-deploy@v1
+      with:
+        subscriptionId: ${{ secrets.AZURE_SUBSCRIPTION }}
+        resourceGroupName: ${{ vars.AZURE_RG }}
+        template: ./.github/workflows/appservice-template.json
+        parameters: acrName=${{ vars.AZURE_ACR_NAME}} appServicePlanName=${{ vars.AZURE_APP_SERVICEPLAN_NAME}} webAppName=${{ vars.AZURE_WEBAPP_NAME}}
```
Comment (CreativePhilip) [inline .github/workflows/deploy.yml:34]: ditto

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+
+    - name: Build Docker image
+      run: |
+        docker build . -t ${{ steps.deploy_acr.outputs.acrLoginServer }}/${{IMAGE_TAG}}:${{ github.sha }}
```
Comment (CreativePhilip) [inline .github/workflows/deploy.yml:41]: ditto

File: .github/workflows/deploy.yml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    - name: Login to ACR
+      uses: azure/docker-login@v1
+      with:
+        login-server: ${{ steps.deploy_acr.outputs.acrLoginServer }}
+        username: ${{ steps.deploy_acr.outputs.registryUsername }}
+        password: ${{ steps.deploy_acr.outputs.registryPassword }}
+
+    - name: Push Docker image to ACR
+      run: |
+        docker push ${{ steps.deploy_acr.outputs.acrLoginServer }}/${{IMAGE_TAG}}:${{ github.sha }}
```
Comment (CreativePhilip) [inline .github/workflows/deploy.yml:52]: ditto

File: .github/workflows/deploy.yml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    - name: Update Web App Image
+      uses: azure/webapps-deploy@v2
+      with:
+        app-name: ${{ vars.AZURE_WEBAPP_NAME }}
+        images: ${{ steps.deploy_acr.outputs.acrLoginServer }}/${{IMAGE_TAG}}:${{ github.sha }}
```
Comment (CreativePhilip) [inline .github/workflows/deploy.yml:58]: ditto

**Review by CreativePhilip — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (jsyska) [reply .github/workflows/deploy.yml]: fixed

**Review by CreativePhilip — APPROVED** [review]

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (jsyska) [reply .github/workflows/deploy.yml]: fixed

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (jsyska) [reply .github/workflows/deploy.yml]: fixed

File: .github/workflows/deploy.yml (hunk 4 above)
Comment (jsyska) [reply .github/workflows/deploy.yml]: fixed

File: .github/workflows/deploy.yml (hunk 5 above)
Comment (jsyska) [reply .github/workflows/deploy.yml]: fixed

**Pull request merged**